hack/lib/golang.sh

```diff
@@ -467,12 +467,17 @@ kube::golang::verify_go_version() {
   # default GO_VERSION to content of .go-version
   GO_VERSION="${GO_VERSION:-"$(cat "${KUBE_ROOT}/.go-version")"}"
   # only setup go if we haven't set FORCE_HOST_GO, or `go version` doesn't match GO_VERSION
-  if ! ([ -n "${FORCE_HOST_GO:-}" ] || \
+  # and GOTOOLCHAIN is auto or unset (the forward thinking idiomatic alternative FORCE_HOST_GO)
+  if [ -n "${FORCE_HOST_GO:-}" ]; then
+    GOTOOLCHAIN="${GOTOOLCHAIN:-local}"
+    export GOTOOLCHAIN
+  elif ! (([ -n "${GOTOOLCHAIN:-}" ] && [ "${GOTOOLCHAIN}" != 'auto' ]) || \
```

Comment:
would a single `"${GOTOOLCHAIN:-auto}" != 'auto'` be more readable?

Reply:
I think that's only slightly confusing if you're not aware that we also intend explicit `"auto"` to be handled the same as unset or set to `""` but we have a comment just above, and it's otherwise easier to read.

Reply:
done